banpei/utils.py

```diff
@@ -33,3 +33,43 @@ def power_method(A, iter_num=1):
     u = Av / s
 
     return u, s, v
+
+
+def _rolling_window(a, window):
+    """
+    Usage:
+    a = np.random.rand(30, 5)
+    for 2d array:
+        roll aling axis=0: rolling_window(a.T, 3).transpose(1, 2, 0)
+        roll along axis=1: rolling_window(a, 3).transpose(1, 0, 2)
+    for 3d array:
+        roll along height(axis=0): rolling_window(a.transpose(2, 1, 0), 3).transpose(2, 3, 1, 0)
+        roll along width(axis=1): rolling_window(a, 3).transpose(2, 0, 1, 3)
+        roll along depth(axis=2): rolling_window(a.transpose(0, 2, 1), 3).transpose(3, 0, 2, 1)
+    """
+    shape = a.shape[:-1] + (a.shape[-1] - window + 1, window)
+    strides = a.strides + (a.strides[-1],)
+    return np.lib.stride_tricks.as_strided(a, shape=shape, strides=strides)
+
+
+def rolling_window(arr, window, axis=0):
```

Comment:
Is this a method you added to extend SST to multidimensional time series data?
I asked because it exceeded the range of PR title `Fixed a bug that may cause TypeError.`.

Reply:
My bad. I forgot I made more modification on my version. Sorry about it. Could you just changed the part of bugfix or whatever you want to add to your current version? Otherwise, I can change it on my end. What should I do next?